### ISSUE TITLE
Can't get review & aggregateRating in JSON-LD metadata to work on live server

### ISSUE THREAD

**Issue by Nellyaa** [nystudio107/craft-seomatic#1247]
### Question
Sorry for bothering, it took two people an entire day and we still couldn't figure out the problem. I hope you can help!
We run Craft (Pro 4.3.3) with Craft Commerce (4.2.1) and SEOmatic (4.0.15). 
We're trying to add the "brand", "aggregateRating" and "review" field to the Commerce product page whose Main Entity of Page is "Product".
For whatever reason our code adds a "brand" field to the JSON-LD no problem, but it's missing the review & aggregateRating on the production/live server. 

### Twig code used

Our product.twig sports the following SEOmatic code:

```
{% set mainEntity = seomatic.jsonLd.get('mainEntityOfPage') %}
{% set brand = seomatic.jsonLd.create({
    'type': 'Brand',
    'name': 'Purrint',
    'url': alias('@web'),
}, false) %}
{% do mainEntity.brand(brand) %}
```

... in which we have an include reviews.twig at a later point whose relevant contents look like this

```
{% set reviews = craft.entries().orderBy('postDate desc').section('reviews').product(product).with(['photo', 'user']).cache().collect() %}
{% set sumratings = 0 %}
{% for review in reviews %}
  {% set sumratings = sumratings + review.rating %}
{% endfor %}

{% if reviews|length %}
    {% set reviewScore = sumratings / reviews|length %}

    {# Add metadata #}
    {% set mainEntity = mainEntity ?? seomatic.jsonLd.get('mainEntityOfPage') %}
    {% set aggregateRating = seomatic.jsonLd.create({
        'type': 'aggregateRating',
        'ratingCount' : reviews|length,
        'reviewCount' : reviews|length,
        'ratingValue': reviewScore|round(1,'common'),
    }, false) %}
    {% do mainEntity.aggregateRating(aggregateRating) %}

    {% set seo_author = seomatic.jsonLd.create({
        'type': 'Person',
        'givenName' : reviews[0].authorName ? reviews[0].authorName : ('verifiedcustomer')|t,
    }, false) %}

    {% set seo_rating = seomatic.jsonLd.create({
        'type': 'rating',
        'ratingValue' : reviews[0].rating,
        'bestRating' : 5,
        'worstRating' : 1,
    }, false) %}

    {% set seo_review = seomatic.jsonLd.create({
        'type': 'review',
        'author' : seo_author,
        'reviewBody' : reviews[0].review,
        'reviewRating' : seo_rating,
    }, false) %}
    {% do mainEntity.review(seo_review) %}

{% endif %}
```

### Object contents

I looked into the mainEntity variable (after the above code) on both environments. Here's the relevant excerpts of aggregateRating and review. Brand looks the same on both but review + aggregateRating differ significantly.
If the page contains no reviews the "review" and "aggregateRating" return null as expected!
I'm really curious why the models are different in both versions and the key and type on the live server is always 'jsonLd' instead of it's correct type.
These two objects are also the only differences I've found between the two environments. Everything else looks the same (like it should). I haven't tested with other objects though.

Return on local:

```
[review] => nystudio107\seomatic\models\jsonld\Review#2
    (
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [include] => true
        [key] => 'review'
        [environment] => null
        [dependencies] => null
        [tagAttrs] => []
        [nonce] => null
        [context] => 'http://schema.org'
        [type] => 'Review'
        [id] => null
        [graph] => null
        [reviewRating] => 4
        [itemReviewed] => null
        [negativeNotes] => null
(( ... The rest omitted for clarity ... ))
    )
```

Return on live server:

```
        [review] => nystudio107\seomatic\models\MetaJsonLd#2
    (
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [include] => true
        [key] => 'jsonLd'
        [environment] => null
        [dependencies] => null
        [tagAttrs] => []
        [nonce] => null
        [context] => 'http://schema.org'
        [type] => 'JsonLd'
        [id] => null
        [graph] => null
    )
```

Return on local: 

```
[aggregateRating] => nystudio107\seomatic\models\jsonld\AggregateRating#3
    (
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [include] => true
        [key] => 'aggregateRating'
        [environment] => null
        [dependencies] => null
        [tagAttrs] => []
        [nonce] => null
        [context] => 'http://schema.org'
        [type] => 'AggregateRating'
        [id] => null
        [graph] => null
        [itemReviewed] => null
        [reviewCount] => 1
        [ratingCount] => 1
        (( ... The rest omitted for clarity ... ))
    )
```

Return on live server:

```
[aggregateRating] => nystudio107\seomatic\models\MetaJsonLd#3
    (
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [include] => true
        [key] => 'jsonLd'
        [environment] => null
        [dependencies] => null
        [tagAttrs] => []
        [nonce] => null
        [context] => 'http://schema.org'
        [type] => 'JsonLd'
        [id] => null
        [graph] => null
    )
```

I hope that helps. Let me know if you need further information of any kind!

**Comment by khalwat** [comment]
I think it might be as simple as this:
```twig
    {% set aggregateRating = seomatic.jsonLd.create({
        'type': 'aggregateRating',
```

The Schema.org types are case-sensitive. Perhaps locally you have a file system that isn't case sensitive, but on production you do.

So what is happening is it tries to find a class that matches that `aggregateRating` but fails to find one on production, and it defaults to the generic `MetaJsonLd` class.

TL;DR: case counts, change it to:

```twig
    {% set aggregateRating = seomatic.jsonLd.create({
        'type': 'AggregateRating',
```

```twig
    {% set seo_rating = seomatic.jsonLd.create({
        'type': 'Rating',
```

```twig
    {% set seo_review = seomatic.jsonLd.create({
        'type': 'Review',
```

Schema.org types are always in `TitleCase` format, though properties are in `camelCase` format.

**Comment by Nellyaa** [comment]
Turns out we were blind indeed! Works as expected. Thank you!
